server/src/platform/drbg/mod.rs

```diff
@@ -0,0 +1,128 @@
+    /// 2. A timestamp of sufficient resolution (detail) so that it is different each time it
+    ///    is used.
+    /// 3. A monotonically increasing sequence number.
+    /// 4. A combination of a timestamp and a monotonically increasing sequence number, such
+    ///    that the sequence number is reset when and only when the timestamp changes. For
+    ///    example, a timestamp may show the date but not the time of day, so a sequence
+    ///    number is appended that will not repeat during a particular day.
+    ///
+    /// For case 1 above, the random value could be acquired from the same source and at the same time
+    /// as the entropy input. For case 2 above, the timestamp must be trusted. A trusted timestamp is
+    /// generated and signed by an entity that is trusted to provide accurate time information.
+    type Nonce: Sized + Default + AsMut<[u8]>;
+    /// Create a new DRBG using the given seed.
+    ///
+    /// All PRNG implementations should be reproducible: given a fixed seed,
+    /// the same sequence of output should be produced on all runs, library
+    /// versions and architectures (e.g. check endianness).
+    fn instantiate(
+        entropy_input: &Self::Entropy,
+        nonce: &Self::Nonce,
+        personalization_string: &[u8],
+    ) -> impl Drbg;
+    /// Add additional entropy to th current RNG
+    ///
+    /// All PRNG implementations should be reproducible: given a fixed seed,
+    /// the same sequence of output should be produced on all runs, library
+    /// versions and architectures (e.g. check endianness).
+    fn reseed(&mut self, entropy_input: &Self::Entropy, additional_input: &[u8]);
+    /// Return the next random [u32].
+    ///
+    /// In the case this method is not implemented directly, it can be
+    /// implemented using `self.next_u64() as u32` or via
+    /// [helpers::next_u32_via_fill].
+    /// # Error
+    /// This function may only fail if the DRBG requires reseeding.
+    fn next_u32(&mut self, additional_input: &[u8]) -> Result<u32, Self::Error>;
+    /// Return the next random [u64].
+    ///
+    /// In the case this method is not implemented directly, it can be
+    /// implemented via [helpers::next_u64_via_u32] or via
+    /// [helpers::next_u64_via_fill].
+    /// # Error
+    /// This function may only fail if the DRBG requires reseeding.
+    fn next_u64(&mut self, additional_input: &[u8]) -> Result<u64, Self::Error>;
+    /// Fill `dest` with random data.
+    ///
+    /// In the case this method is not implemented directly, it can be
+    /// implemented via [helpers::fill_bytes_via_next].
+    ///
+    /// This method should guarantee that `dest` is entirely filled with
+    /// new data.
+    ///
+    /// # Error
+    /// This function may fail in one of two cases:
+    /// 1. Requested output is too large
+    /// 2. DRBG requires reseeding.
+    fn fill_bytes(&mut self, additional_input: &[u8], dest: &mut [u8]) -> Result<(), Self::Error>;
```

Comment:
You could actually provide a default blanket implementation here that does relatively the right thing by calling `next_u64` and all that. That leaves it open for others to provide manually if they have a better way of doing it, but also doesn't make this a required function if the other two calls are sufficient.

Reply:
the idea is what if the actual hardware implementation provides `fill_bytes` for instance nist standards provide only `fill_bytes` (which will be clearer in the next few PR,

But hardware instruction that fills arbitrary buffer is impractical so there is good chance that different scenarios would require different defaults.